File: base/src/main/java/vproxy/base/util/coll/Tuple3.java
package vproxy.base.util.coll;

public class Tuple3<A, B, C> {
    public final A _1;
    public final B _2;
    public final C _3;

    public Tuple3(A _1, B _2, C _3) {
        this._1 = _1;
        this._2 = _2;
        this._3 = _3;
    }

    @Override
    public String toString() {
        return "Tuple(" + _1 + ", " + _2 + ", " + _3 + ")";
    }
}

File: base/src/main/kotlin/io/vproxy/base/util/coll/Tuple3.kt
package io.vproxy.base.util.coll

data class Tuple3<A, B, C>(
  @JvmField
  val _1: A,
  @JvmField
  val _2: B,
  @JvmField
  val _3: C
) {
  override fun toString(): String {
    return "Tuple($_1, $_2, $_3)"
  }
}
